kratos/modeler/connectivity_preserve_modeler.h

```diff
@@ -162,6 +170,36 @@ class KRATOS_API(KRATOS_CORE) ConnectivityPreserveModeler : public Modeler
         ModelPart& rDestinationModelPart
     ) const;
 
+    template <typename TContainer>
+    long unsigned int GetMaxId(
+        const TContainer& rContainer,
+        const Communicator& rCommunicator
+    ) const
+    {
+        KRATOS_TRY
+
```

Comment:
I am not sure if this is required ... the containers of elements, conditions and nodes are all `PointerVectorSet` s , 
This means that they are always sorted in ascending order (if properly used). This would mean that 
`(rContainer.end()-1)->Id()` will give you the max Id in that container already.

Reply:
I will change it as you suggested, if @RiccardoRossi approves the existing conceptual changes :)